signer/src/testing/request_decider.rs

```diff
@@ -146,14 +144,18 @@ where
 {
     /// Assert that the transaction signer will make and store decisions
     /// for pending deposit requests.
-    pub async fn assert_should_store_decisions_for_pending_deposit_requests(self) {
-        let mut rng = get_rng();
+    pub async fn assert_should_store_decisions_for_pending_deposit_requests<
+        R: rand::Rng + rand::CryptoRng + Send + Clone + 'static,
+    >(
+        self,
+        rng: &mut R,
+    ) {
```

Comment:
For large function signatures like this, can we format it like this here?
```suggestion
    pub async fn assert_should_store_decisions_for_pending_deposit_requests<R>(
        self,
        rng: &mut R,
    )
    where
        R: rand::Rng + rand::CryptoRng + Send + Clone + 'static 
    {
```